Build failure: ecm

## Steps To Reproduce

Steps to reproduce the behavior:

With nixos-unstable as the main nixpkgs and `labplot` in home-manager packages (or systemPackages, should not matter) -- ecm is a transitive dependency:

```
       > checking if globals are prefixed by underscore... configure: error: Test program links neither with nor without underscore.
       For full logs, run 'nix log /nix/store/svr8i4l2fcw6ihhfn1s2r70fwn92x7bh-ecm-7.0.4.drv'.
error: 1 dependencies of derivation '/nix/store/8vpsqwyqbivz25p5nqx8ir4xqzzkbqxa-sage-with-env-10.5.drv' failed to build
error: 1 dependencies of derivation '/nix/store/cklyrnxqdc3acg3wh2mlf5ci47kyf3ng-cantor-23.08.5.drv' failed to build
error: 1 dependencies of derivation '/nix/store/9188d36jy6jyyvq3xqxnlkc7k1njhmf1-labplot-2.10.1.drv' failed to build
```

1. nixos-rebuild --use-remote-sudo --flake foo#bar boot

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
$ nix log /nix/store/svr8i4l2fcw6ihhfn1s2r70fwn92x7bh-ecm-7.0.4.drv
warning: The interpretation of store paths arguments ending in `.drv` recently changed. If this command is now failing try again with '/nix/store/svr8i4l2fcw6ihhfn1s2r70fwn92x7bh-ecm-7.0.4.drv^*'
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/rxpfyf3d9145z87vpz97ilaa4dfqwzy6-ecm-7.0.4.tar.gz
source root is ecm-7.0.4
setting SOURCE_DATE_EPOCH to timestamp 1476178092 of file ecm-7.0.4/ecm.h
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Updating Autotools / GNU config script to a newer upstream version: ./config.sub
Updating Autotools / GNU config script to a newer upstream version: ./config.guess
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
fixing libtool script ./ltmain.sh
./configure
patching script interpreter paths in ./configure
./configure: interpreter directive changed from "#! /bin/sh" to "/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/sh"
configure flags: --disable-static --disable-dependency-tracking --prefix=/nix/store/0vbj17454w6f2x6d1z27ibrwjmy1sdab-ecm-7.0.4
checking for a BSD-compatible install... /nix/store/4s9rah4cwaxflicsk5cndnknqlk9n4p3-coreutils-9.5/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /nix/store/4s9rah4cwaxflicsk5cndnknqlk9n4p3-coreutils-9.5/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking for grep that handles long lines and -e... /nix/store/aap6cq56amx4mzbyxp2wpgsf1kqjcr1f-gnugrep-3.11/bin/grep
checking for egrep... /nix/store/aap6cq56amx4mzbyxp2wpgsf1kqjcr1f-gnugrep-3.11/bin/grep -E
checking for a sed that does not truncate output... /nix/store/abm77lnrkrkb58z6xp1qwjcr1xgkcfwm-gnused-4.9/bin/sed
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking for style of include used by make... GNU
checking dependency style of gcc... none
checking dependency style of gcc... none
checking how to print strings... printf
checking for a sed that does not truncate output... (cached) /nix/store/abm77lnrkrkb58z6xp1qwjcr1xgkcfwm-gnused-4.9/bin/sed
checking for fgrep... /nix/store/aap6cq56amx4mzbyxp2wpgsf1kqjcr1f-gnugrep-3.11/bin/grep -F
checking for ld used by gcc... ld
checking if the linker (ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... nm
checking the name lister (nm) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
checking for ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... (cached) pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse nm output from gcc object... ok
checking for sysroot... no
checking for a working dd... /nix/store/4s9rah4cwaxflicsk5cndnknqlk9n4p3-coreutils-9.5/bin/dd
checking how to truncate binary pipes... /nix/store/4s9rah4cwaxflicsk5cndnknqlk9n4p3-coreutils-9.5/bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking how to run the C preprocessor... gcc -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... no
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... no
checking whether to build static libraries... yes
checking for int64_t... yes
checking for uint64_t... yes
checking for unsigned long long int... yes
checking for long long int... yes
checking for an ANSI C-conforming const... yes
checking for inline... inline
checking whether time.h and sys/time.h may both be included... yes
checking for size_t... yes
checking if assembly code is ELF... yes
checking for suitable m4... m4
checking how to switch to text section... .text
checking how to export a symbol... .globl
checking what assembly label suffix to use... :
checking if globals are prefixed by underscore... configure: error: Test program links neither with nor without underscore.
```

</details>

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->
 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.5, NixOS, 25.05 (Warbler), 25.05.20241219.d70bd19`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.91.1
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux, x86_64-v4-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/user/.config/nix/nix.conf:/nix/store/qd1p658xdl2wl6hqk01n6gcphhka8snr-plasma-workspace-6.2.4/etc/xdg/nix/nix.conf:/nix/store/7zl907185jz528xzmifaarnqkyrkdx13-kglobalacceld-6.2.4/etc/xdg/nix/nix.conf:/nix/store/m3jp6qy9asm38m8l5cnryck7c5akfivw-baloo-6.8.0/etc/xdg/nix/nix.conf:/home/user/.config/kdedefaults/nix/nix.conf:/nix/store/qd1p658xdl2wl6hqk01n6gcphhka8snr-plasma-workspace-6.2.4/etc/xdg/nix/nix.conf:/nix/store/7zl907185jz528xzmifaarnqkyrkdx13-kglobalacceld-6.2.4/etc/xdg/nix/nix.conf:/nix/store/m3jp6qy9asm38m8l5cnryck7c5akfivw-baloo-6.8.0/etc/xdg/nix/nix.conf:/nix/store/qd1p658xdl2wl6hqk01n6gcphhka8snr-plasma-workspace-6.2.4/etc/xdg/nix/nix.conf:/nix/store/7zl907185jz528xzmifaarnqkyrkdx13-kglobalacceld-6.2.4/etc/xdg/nix/nix.conf:/nix/store/m3jp6qy9asm38m8l5cnryck7c5akfivw-baloo-6.8.0/etc/xdg/nix/nix.conf:/home/user/.config/kdedefaults/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/user/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/user/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/user/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/873hdnqclw6c5kqax9mmw57a72rn4c3h-lix-2.91.1/share`
 - nixpkgs: `/nix/store/8fwsiv0hd7nw1brkvka0jf1frk3m7qkr-source`

## Notify maintainers

@roconnor 

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ec/ecm/package.nix
+++ b/pkgs/by-name/ec/ecm/package.nix
@@ -8,7 +8,7 @@
 
 let
   pname = "ecm";
-  version = "7.0.4";
+  version = "7.0.6";
   name = "${pname}-${version}";
 in
 
@@ -16,9 +16,15 @@ stdenv.mkDerivation {
   inherit name;
 
   src = fetchurl {
-    url = "http://gforge.inria.fr/frs/download.php/file/36224/ecm-${version}.tar.gz";
-    sha256 = "0hxs24c2m3mh0nq1zz63z3sb7dhy1rilg2s1igwwcb26x3pb7xqc";
+    url = "https://gitlab.inria.fr/zimmerma/ecm/uploads/ad3e5019fef98819ceae58b78f4cce93/ecm-${version}.tar.gz";
+    hash = "sha256-fSDs5hq2ogrYXywYBkyr133Eapb/iUtSINuxbkZm6KU=";
   };
+
+  postPatch = ''
+    patchShebangs test.ecmfactor
+    patchShebangs test.ecm
+    substituteInPlace test.ecm --replace /bin/rm rm
+  '';
 
   # See https://trac.sagemath.org/ticket/19233
   configureFlags = lib.optional stdenv.hostPlatform.isDarwin "--disable-asm-redc";
@@ -33,8 +39,8 @@ stdenv.mkDerivation {
   meta = {
     description = "Elliptic Curve Method for Integer Factorization";
     mainProgram = "ecm";
-    license = lib.licenses.gpl2Plus;
-    homepage = "http://ecm.gforge.inria.fr/";
+    license = lib.licenses.gpl3Only;
+    homepage = "https://gitlab.inria.fr/zimmerma/ecm";
     maintainers = [ lib.maintainers.roconnor ];
     platforms = with lib.platforms; linux ++ darwin;
   };

